backend/plugin/db/mysql/query.go

```diff
@@ -55,3 +58,48 @@ func getStatementWithResultLimitForTiDB(singleStatement string, limitCount int)
 	}
 	return "", nil
 }
+
+// singleStatement must be a selectStatement for mysql.
+func getStatementWithResultLimitForMySQL(singleStatement string, limitCount int) (string, error) {
+	list, err := mysqlparser.ParseMySQL(singleStatement)
+	if err != nil {
+		return "", err
+	}
+
+	listener := &mysqlRewriter{
+		limitCount:     limitCount,
+		outerMostQuery: true,
+	}
+
+	for _, stmt := range list {
+		listener.rewriter = *antlr.NewTokenStreamRewriter(stmt.Tokens)
+		antlr.ParseTreeWalkerDefault.Walk(listener, stmt.Tree)
+		if listener.err != nil {
+			return "", errors.Wrapf(listener.err, "statement: %s", singleStatement)
+		}
+	}
+	return listener.rewriter.GetTextDefault(), nil
+}
+
+type mysqlRewriter struct {
+	*mysql.BaseMySQLParserListener
+
+	rewriter       antlr.TokenStreamRewriter
+	err            error
+	outerMostQuery bool
+	limitCount     int
+}
+
+func (r *mysqlRewriter) EnterQueryExpression(ctx *mysql.QueryExpressionContext) {
+	if !r.outerMostQuery {
+		return
+	}
+	r.outerMostQuery = false
+	limitClause := ctx.LimitClause()
+	if limitClause != nil {
+		// limit clause already exists.
+		return
+	}
```

Comment:
Should we consider the following case:

`SELECT * FROM t LIMIT 10000000`

Do we need to rewrite it to `LIMIT 10`?